### PR TITLE
TC-IDM-4.2 Step 8 delay adjust

### PR DESCRIPTION
**Pull request by raul-marquez-csa**
Adds 1/2 second to existing delay in step 8 before reading updated attribute value

From:
https://github.com/project-chip/connectedhomeip/pull/31113

### CODE REVIEW

File: src/python_testing/TC_IDM_4_2.py [hunk 1]
```diff
@@ -409,7 +409,7 @@ async def test_TC_IDM_4_2(self):
         )
 
         # Wait MinIntervalFloor seconds before reading updated attribute value
-        time.sleep(same_min_max_interval_sec)
+        time.sleep(same_min_max_interval_sec + 0.5)
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_IDM_4_2.py:412]: Why 0.5? Why not 0.2 or 1.0? Also, this adds more time than the min interval. If this is failing, then changing the delay is NOT the solution and the test case should be revisited to be more timing-insensitive if possible.

**Review by tcarmelveilleux — CHANGES_REQUESTED** [review]
This needs to be revisited at the test plan level.

File: src/python_testing/TC_IDM_4_2.py (hunk 1 above)
Comment (tcarmelveilleux) [reply src/python_testing/TC_IDM_4_2.py:412]: This does need a test plan update. @cecille working on it.

Comment (raul-marquez-csa) [reply src/python_testing/TC_IDM_4_2.py]: Added loop

**Comment by raul-marquez-csa** [comment]
Basically the test in the CI was timing out due to performing a read before the value was available due to the different response times in different network environments.

The last update implements a loop that times out in 10 seconds that checks for the value in increments of 0.1 seconds until it receives the value.

This is a temporary patch until @cecille works out the proper solution.

File: src/python_testing/TC_IDM_4_2.py [hunk 2]
```diff
@@ -67,6 +67,29 @@ async def get_idle_mode_duration_sec(self, ctrl, ep=ROOT_NODE_ENDPOINT_ID):
...
+        increment = 0.1
+        loop = True
+        attribute_value = None
+        while loop:
+            # Get the attribute value
+            attribute_value = sub.GetAttribute(typed_attr_path)
+
+            # Check if the value is not an empty string
+            if attribute_value != "":
+                loop = False  # Exit the loop
```
Comment (cecille) [inline src/python_testing/TC_IDM_4_2.py:82]:
```suggestion
                return attribute_value
```
Then you can get rid of the loop var and just loop on time.time() < end_time and throw the error at the end

File: src/python_testing/TC_IDM_4_2.py [hunk 3]
```diff
@@ -67,6 +67,29 @@ async def get_idle_mode_duration_sec(self, ctrl, ep=ROOT_NODE_ENDPOINT_ID):
             attribute=Clusters.IcdManagement.Attributes.IdleModeDuration
         )
 
+    def get_attribute_value_wait(self, sub, typed_attr_path):
+        start_time = time.time()
+        timeout = 10
+        increment = 0.1
```
Comment (andy31415) [inline src/python_testing/TC_IDM_4_2.py:73]: I think we may have a few of these sprinkled in the code. How about adding a retry library in the testing requirements like https://tenacity.readthedocs.io/en/latest/ or maybe there are others, so we can read code like `@retry(stop=stop_after_delay(10), wait=wait_fixed(0.1), retry=retry_if_result(is_empty)))` or something like that.

Comment (cecille) [reply src/python_testing/TC_IDM_4_2.py:73]: ugh...I hope we don't. If we do, that seems like a test red flag.

Comment (cecille) [reply src/python_testing/TC_IDM_4_2.py:73]: I put up a PR to disable this test in the CI. The test plan has been updated so we don't have to do this. I'm hoping we can just skip this retry at this point and fix the test properly. This was meant to be a quick fix-forward to get this to stop flaking.

https://github.com/project-chip/connectedhomeip/pull/32378

**Pull request closed without merging**